packages/node_modules/@webex/internal-plugin-devices/test/unit/spec/interceptors/device-url.js

```diff
@@ -0,0 +1,177 @@
+      describe('when `cisco-device-url` header is are already set', () => {
+        it('returns the options', () => {
+          interceptor.webex.internal.device.url = undefined;
+          options = {
+            headers: {
+              'cisco-device-url': fixture.url
+            },
+            ...options
+          };
+
+          interceptor.onRequest(options)
+            .then((results) => assert.equal(results, options));
+        });
+      });
+
+      describe('adds cisco-device-url to the request header for service catalog services', () => {
+        it('when only the service property is provided', () => {
+          options.service = fixture.service;
+
+          interceptor.onRequest(options)
+            .then((results) => {
+              assert.calledWith(
+                waitForService,
+                {
+                  service: options.service,
+                  url: undefined
+                }
+              );
+              assert.calledWith(
+                getServiceFromUrl,
+                fixture.serviceUrl
+              );
+              assert.isDefined(options.headers['cisco-device-url']);
+              assert.equal(results.headers['cisco-device-url'], fixture.uri);
+            });
+        });
+
+        it('when only the uri property is provided', () => {
+          options = {
+            uri: fixture.uri,
+            ...options
+          };
+
+          interceptor.onRequest(options)
+            .then((results) => {
+              assert.calledWith(
+                waitForService,
+                {
+                  service: undefined,
+                  url: options.uri
+                }
+              );
+              assert.calledWith(
+                getServiceFromUrl,
+                fixture.serviceUrl
+              );
+              assert.isDefined(results.headers['cisco-device-url']);
+              assert.equal(results.headers['cisco-device-url'], fixture.uri);
+            });
+        });
+
+        it('when both the service and uri properties are provided', () => {
+          options = {
+            services: fixture.service,
+            uri: fixture.uri,
+            ...options
+          };
+
+          interceptor.onRequest(options)
+            .then((results) => {
+              assert.calledWith(
+                waitForService,
+                {
+                  service: options.service,
+                  url: options.uri
+                }
+              );
+              assert.calledWith(
+                getServiceFromUrl,
+                fixture.serviceUrl
+              );
+              assert.isDefined(results.headers['cisco-device-url']);
+              assert.equal(results.headers['cisco-device-url'], fixture.uri);
+            });
+        });
+
+        describe('does not add cisco-device-url due to invalid service name', () => {
+          beforeEach(() => {
+            getServiceFromUrl.returns({name: 'saml'});
```

Comment:
Looks great! we might want to add assertions for the other service names in this section.